sympy/vector/functions.py

```diff
@@ -474,3 +474,56 @@ def _path(from_object, to_object):
         from_path.append(other_path[i])
         i -= 1
     return index, from_path
+
+def orthogonalize(*vlist, **kwargs):
+    """
+    Takes a list of independent vectors and orthogonalizes them
+    using the Gram - Schmidt process. Returns a list of
+    orthogonal or orthonormal vectors.
+
+    Parameters
+    ==========
+
+    vlist : list of independent vectors to be made orthogonal.
+
+    orthonormal : Optional parameter
+    Set to True if the the vectors returned should be orthonormal.
+    Default: False
+
+    Examples
+    ========
+
+    >>> from sympy.vector.coordsysrect import CoordSysCartesian
+    >>> from sympy.vector.vector import Vector, BaseVector
+    >>> from sympy.vector.functions import orthogonalize
+    >>> C = CoordSysCartesian('C')
+    >>> i, j, k = C.base_vectors()
+    >>> v1 = i + 2*j
+    >>> v2 = 2*i + 3*j
+    >>> orthogonalize(v1, v2)
+    [C.i + 2*C.j, 2/5*C.i + (-1/5)*C.j]
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Gram-Schmidt_process
+
+    """
+    orthonormal = kwargs.get('orthonormal', False)
+
+    if not all(isinstance(vec, Vector) for vec in vlist):
+        raise TypeError('Each element must be of Type Vector')
+
+    ortho_vlist = []
+    for i in range(len(vlist)):
+        term = vlist[i]
+        for j in range(i):
+            term -= ortho_vlist[j].vec_project(vlist[i])
+        if term == Vector.zero:
```

Comment:
@moorepants, if the vectors are linearly dependent, then the Gram-Schmidt process will output a zero-vector at the ith step assuming vi is a linear combination of v0, v1, v2, ...., vi-1. This is because, if vi is a linear combination of the previous vectors, the formula for the ith vector according to the Gram - Schmidt process will result in a zero vector. This is explained in 
https://en.wikipedia.org/wiki/Gram%E2%80%93Schmidt_process 
I have added a test for it (line 175). 
However, an alternative would be to allow a linearly dependent sequence of vectors and remove zero vectors from the final list of vectors. The number of vectors output by the algorithm would then be equal to the dimension of the space spanned by the original input. Should I go ahead and do that ?